testsuite/tests/libratest/smoke_test.rs

```diff
@@ -930,6 +930,72 @@ fn test_e2e_reconfiguration() {
     );
 }
 
+#[test]
+fn test_e2e_modify_publishing_option() {
+    let (mut env, mut client_proxy) = setup_swarm_and_client_proxy(1, 0);
+    client_proxy.create_next_account(false).unwrap();
+
+    client_proxy
+        .mint_coins(&["mintb", "0", "10"], true)
+        .unwrap();
+    assert_eq!(
+        Decimal::from_f64(10.0),
+        Decimal::from_str(&client_proxy.get_balance(&["b", "0"]).unwrap()).ok()
+    );
+    let script_path = workspace_builder::workspace_root()
+        .join("testsuite/tests/libratest/dev_modules/test_script.mvir");
+    let unwrapped_script_path = script_path.to_str().unwrap();
+    let script_params = &["execute", "0", unwrapped_script_path, "script"];
+    let script_compiled_path = client_proxy.compile_program(script_params).unwrap();
+
+    // Initially publishing option was set to CustomScript, this transaction should be executed.
+    client_proxy
+        .execute_script(&["execute", "0", &script_compiled_path[..], "10", "0x0"])
+        .unwrap();
+
+    assert_eq!(
+        client_proxy
+            .get_sequence_number(&["sequence", "0", "true"])
+            .unwrap(),
+        1
+    );
+
+    client_proxy
+        .disallow_custom_script(&["disallow_custom_script"], true)
+        .unwrap();
+
+    // TODO: Currently VMValidator didn't restart after reconfiguration. We will manually restart
+    //       the node so that VMValidator is using the new config.
```

Comment:
Even though `VMValidator` will let the bad script into mempool, shouldn't trying to actually execute result in an abort? I guess I understand why we would need to do the restart if we were enabling custom scripts, but it doesn't seem like we should need it for disabling.

Reply:
The problem is as following: it's a bit hard to prove that this transaction is NOT committed. So if we just let the transaction through, the client will actually panic with a timeout as this transaction is never committed.

Reply:
I'm confused. Shouldn't this transaction be committed regardless of what the script setting is? In my understanding, the only difference is that we'll see an abort if custom scripts is disabled (`UNKNOWN_SCRIPT`) and a successfully executed transaction if custom scripts is enabled.

Reply:
There's some slight difference here:
In the case when VMValidator didn't restart, this transaction will pass the VMValidator. Thus the request to submit a transaction will return with a success status. This transaction will then flow to Executor and will be dropped there. The client here will thus have no idea if this transaction is indeed dropped as this UNKNOWN_SCRIPT error will not be persisted and cannot be queried.

In the case when VMValidator restart, however, the VMValidator will reject this transaction, thus the request to submit a transaction will return with a UNKNOWN_SCRIPT error.

Reply:
Ah, thanks for the clarification! I think my confusion results from the following:

Why does the executor discard an `UNKNOWN_SCRIPT` transaction?

Reply:
Ah, I see that my expectations are wrong and `UNKNOWN_SCRIPT` is actually a `Discard`. Will look into fixing this separately. Your explanation makes sense, thanks!